### ISSUE TITLE
Registration

### ISSUE THREAD

**Issue by malybobo** [haxball/haxball-issues#347]
Hey,
Basro, is it possible to enter account registration?
Of course, haxball will be further available to unregistered players.
But for example, the player who is registered his nickname is gray
and the one that is not registered remains white as it was before.
Who wants to register - optional registration.
this will also help to identify the player.

I do not quite understand the idea that the current player would only be assigned an id.
What will happen if the new operating system is installed? will clear the cache?
Will he still have the same id or new?

The vast majority of regular players are in agreement for an optional account registration.

I play hax almost from the beginning.
I am also missing registration. A button at the bottom of the body "if you want to register"

This will help with silencing a given player, excluding him from the chat, perpetual ban or permanent admin.

From the technical side, it seems very easy for you.

Please, think about my request.

**Comment by Leo-MapCreator** [comment]
Its better to the registered player color to be white instead of gray, and the unregistrered player to be gray instead of white

**Comment by malybobo** [comment]
Basro, please tell me if you plan to register in the future

**Comment by Leo-MapCreator** [comment]
I think he does @malybobo

**Comment by Leo-MapCreator** [comment]
@saviola777 it prevents a lot though

**Comment by saviola777** [comment]
That's not true for Germany at least. If I turn off my modem and turn in on again, I get a new IP. And many ISPs, especially in the rural areas, still employ 24 hour disconnects, so people get a new IP every day. I don't think automatic identification without user interaction is a viable solution, because changing browsers, IPs etc. will reset a user's identity (similar to TeamSpeak – but there you have admins who can help re-acquire lost permissions etc., which would not be possible for Haxball).

I think identification could be handled similar to how it's done on IRC, including optional registration/identification (authenticated people are marked in a certain way), ghost kicking etc. This would not disrupt people playing public, while adding a way for leagues etc. to enforce authentication.

But on the other hand, I don't think this is an essential feature. And it could be implemented on top of the headless API by the community if people feel it's really required.

**Comment by malybobo** [comment]
366/5000
In my opinion basro does not want to use the server for these tasks.
Checking for ip is pointless - in Poland, the majority of homes are also variable ip.
I do not know why basro did not take a sentence in this thread.
As for id, I understand the idea of this solution, but haxball players are too lazy.
Even if basro does not make changes, haxbll still rules,
Thanks.
